libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt

```diff
@@ -0,0 +1,8 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+
+interface FasterRouteObserver {
+
+    fun onFasterRouteAvailable(fasterRoute: DirectionsRoute)
```

Comment:
How about we provide a list of routes from the response here?

Reply:
> ```
> private fun requestFasterRoute(routeOptions: RouteOptions) {
>        if (routes.isEmpty()) {
>            return
>        }
>        router.getRoute(routeOptions, object : Router.Callback {
>            override fun onResponse(routes: List<DirectionsRoute>) {
>                val route = routes[0]
>                if (isRouteFaster(route)) {
>                    fasterRouteObservers.forEach { it.onFasterRouteAvailable(route) }
>                }
>            }
>
>            override fun onFailure(throwable: Throwable) {
>                // do nothing
>            }
>
>            override fun onCanceled() {
>                // do nothing
>            }
>        })
>    }
> ```

If you look at the `onResponse` we use the route at index 0 to check if the route is faster than the current route. If true only then we broadcast a faster route. The question, I would like to ask here is do we want to emit a list of routes because then how does the end client know which route is faster in that list? There are 2 solutions to this:
* We emit the list of route and say that route at index 0 in this list has been verified to be faster; so use it - so then in that case why to even send a list since the 0th route is always going to be faster. 
* Send a list to the user and ask him to find out which route is faster

I don't see either of the solution that we would probably want to extend from SDK perspective. 

Faster route in itself is a very specific feature that says we found 1 faster route rather than multiple faster routes. In that case I think we should only emit one route.
Comments/Thoughts?